src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php

```diff
@@ -69,8 +69,10 @@ public function createConnectionByQueueConnectionTransfer(QueueConnectionTransfe
      */
     protected function createOrGetConnection(QueueConnectionTransfer $queueConnectionTransfer): ConnectionInterface
     {
-        if (isset($this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()])) {
-            return $this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()];
+        $connection = $this->createdConnectionsByConnectionName[$queueConnectionTransfer->getName()] ?? null;
+
+        if ($connection !== null && $connection->getChannel() !== null && $connection->getChannel()->is_open()) {
```

Comment:
I would move this condition to the separate method with a meaningful name

Reply:
Also changelog says that we made it to use cached connection, but it was original behaviour too. I assume we want to use a **working** cached connection

Reply:
fixed